lib/livebook/runtime/erl_dist/runtime_server.ex

```diff
@@ -338,7 +346,7 @@ defmodule Livebook.Runtime.ErlDist.RuntimeServer do
       opts
     )
 
-    {:noreply, state}
+    {:noreply, Map.replace!(state, :last_evaluator, state.evaluators[container_ref])}
```

Comment:
```suggestion
    {:noreply, %{state | last_evaluator: state.evaluators[container_ref]}
```

Also, would be worth extracting it into a variable `evaluator = state.evaluators[container_ref]` :)